Rules/ReviewUnusedParameter.cs

```diff
@@ -72,6 +72,34 @@ public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
             }
         }
 
+        /// <summary>
+        /// Checks for PS bound parameter reference.
+        /// </summary>
+        /// <param name="ast"></param>
+        /// <returns></returns>
+        private static bool IsBoundParametersReference(Ast ast)
+        {
+            bool isBound = false;
+            // if $MyInvocation.BoundParameters or $PSCmdlet.MyInvocation.BoundParameters used.
+            if (ast is MemberExpressionAst memberAst
+                && memberAst.Member is StringConstantExpressionAst memberStringAst
+                && memberStringAst.Value.Equals("BoundParameters", StringComparison.OrdinalIgnoreCase))
+            {
+                // handle both regular and nested scenarios.
+                isBound = (memberAst.Expression is VariableExpressionAst veAst
+                    && veAst.VariablePath.UserPath.Equals("MyInvocation", StringComparison.OrdinalIgnoreCase))
+                    || (memberAst.Expression is MemberExpressionAst meAst
+                    && meAst.Expression is VariableExpressionAst veAstNested
+                    && veAstNested.VariablePath.UserPath.Equals("PSCmdlet", StringComparison.OrdinalIgnoreCase)
+                    && meAst.Member is StringConstantExpressionAst sceAstNested
+                    && sceAstNested.Value.Equals("MyInvocation", StringComparison.OrdinalIgnoreCase));
+            }
+
+            return isBound
+                || ast is VariableExpressionAst variableAst

```

Comment:
This whole method is pretty hard to follow.  I would recommend re-writing it to be a clearer set of conditions.

```
        private static bool IsBoundParametersReference(Ast ast)
        {
            if (!(ast is VariableExpressionAst variableAst) || 
                variableAst.VariablePath.UserPath.Equals("PSBoundParameters", StringComparison.OrdinalIgnoreCase))
            {
                 return false;
            }

            // and so on - with ifs using no more than 3 conditions.

        }
```

Reply:
I hope 3 condition is not a hard limit? I updated this function to have readable if conditions. The condition for checking $PSCmdlet.MyInvocation still has more than 3 conditions (i.e 4).